lib/generators/gql/templates/input_type.rb

```diff
@@ -1,9 +1,9 @@
 module Types
   module Input
-    class <%= @model_name %>Input < Types::BaseInputObject
+    class <%= @name %>Input < <%= @superclass %>
```

Comment:
I think the `Input` suffix should only apply to the default name. Thoughts? 

Personally I never use it. It's redundant. The namespace starts with `Types`.

Reply:
Another thought (for possibly another PR): If one says: `--name Some::Foo::Bar` _those modules_ are generated instead of `Types::Input`

Or, maybe if one says `Some::Foo::Bar` we generate `Types::Input::Some::Foo::Bar` and of they do `::Some::Foo::Bar` we wipeout `Types::Input`?

Reply:
> I think the `Input` suffix should only apply to the default name. Thoughts?
> 
> Personally I never use it. It's redundant. The namespace starts with `Types`.

I think this is what graphql-ruby does by default, presumably mimicking the rails defaults (e.g. PostController). I'm good with the ability to override it but I tend to like it as a default, and find it helpful for differentiating inputs from types.

Reply:
> Another thought (for possibly another PR): If one says: `--name Some::Foo::Bar` _those modules_ are generated instead of `Types::Input`
> 
> Or, maybe if one says `Some::Foo::Bar` we generate `Types::Input::Some::Foo::Bar` and if they do `::Some::Foo::Bar` we wipeout `Types::Input`?

My initial thought is that the `::` is a subtle detail for such a substantive change, but on the other hand you're probably only going to use it if you want. I don't have much of an opinion on this because I just use the graphql-ruby defaults, so I'm good with this change.

Reply:
> I think the Input suffix should only apply to the default name. Thoughts?

> > I'm good with the ability to override it but I tend to like it as a default, and find it helpful for differentiating inputs from types.

Ok, let's keep `Type` regardless and see where it goes. Maybe I'm the only one who doesn't use this convention so no need to change just for me 😏

Reply:
Just ended up adding a `--namespace` option. Better than parsing `--name`, I think. 

With namespace support (which I only added for 2 generators) it makes more sense to build the code outside ERB.